### ISSUE TITLE
SLA file is missing

### ISSUE THREAD

**Issue by Elmue** [esaulenka/Ghidra_C166#9]
I load your files into a new subfolder
C:\Program Files (x86)\Ghidra\Ghidra\Processors\C166\
and Ghidra complains that the SLA file is missing.

java.io.FileNotFoundException: C:\Program Files (x86)\Ghidra\Ghidra\Processors\C166\data\languages\c166.sla

All other processors have a SLA file.
What is wrong here ?

**Comment by esaulenka** [comment]
sorry for the late answer

Ghidra should automatically compile SLA from SLASPEC files on first run. Looks like it doesnt have access rights to write new files to Program Files folder.
Also you can manually compile SLA with Program Files\Ghidra\support\sleigh.bat

But please note, that this module not completed yet. Main issue is EXTP/EXTR/etc instructions not working.

**Comment by Elmue** [comment]
Thanks for your answer.
I changed  the directory permissions and  now it works.

**Comment by GhidrAuto** [comment]
elmu, how did you grant permission? I think I know how, but it is not working for me. different processor, but it is having issues, supposedly if I delete the SLA and then start ghidra it will fix it, however, it is not creating a new SLA

**Comment by Elmue** [comment]
> elmu, how did you grant permission?

I cannot believe that you want to use Ghidra (which supposes that you are an advanced cracker) and you don't know how to set the permissions of a folder in Windows!

Use Google to find how you set folder permissions!

**Comment by GhidrAuto** [comment]
Tell me about it! I discovered Ghidra 5 months ago, learning curve has been brutal. Here's my YT channel on automotive ECU disassembly https://youtu.be/jRo1wr7o80g?si=WXrUi5MLlD1-Xo3y. feel free to drop in and drop some knowledge, or shame lol. Ghidra is definitely a powerful tool. yeah, I followed the instructions I found by googling. it still isn't creating an .SLA. Trying to figure out what is going wrong.

tHaNks for the help CrAckEr

**Comment by GhidrAuto** [comment]
> Tell me about it! I discovered Ghidra 5 months ago, learning curve has been brutal. Here's my YT channel on automotive ECU disassembly https://youtu.be/jRo1wr7o80g?si=WXrUi5MLlD1-Xo3y. feel free to drop in and drop some knowledge, or shame lol. Ghidra is definitely a powerful tool. yeah, I followed the instructions I found by googling. it still isn't creating an .SLA. Trying to figure out what is going wrong.
> 
> tHaNks for the help CrAckEr

**Comment by GhidrAuto** [comment]
> > elmu, how did you grant permission?
> 
> I cannot believe that you want to use Ghidra (which supposes that you are an advanced cracker) and you don't know how to set the permissions of a folder in Windows!
> 
> Use Google to find how you set folder permissions!

Tell me about it! I discovered Ghidra 5 months ago, learning curve has been brutal. Here's my YT channel on automotive ECU disassembly https://youtu.be/jRo1wr7o80g?si=WXrUi5MLlD1-Xo3y. feel free to drop in and drop some knowledge, or shame lol. Ghidra is definitely a powerful tool. yeah, I followed the instructions I found by googling. it still isn't creating an .SLA. Trying to figure out what is going wrong.

Thanks for the help Cracker

**Comment by esaulenka** [comment]
Hi @GhidrAuto ! Haha, this world is a small enough ;-)

I advice you to install Ghidra in your work folder, not in common `Program Files` (as I understood from your videos, you are on Windows).
Also it can be useful to try to compile `.slaspec` files with `Ghidra\support\sleigh.bat` - it is a commandline tool, and it can wrote some useful info if something going wrong.

But please note that this module still in abandoned state, and I haven't found a good way to support such confusing memory addressing (almost all 16-bit MCUs is a big pain).

**Comment by Elmue** [comment]
Here I send you my project for the C166 processor.
Copy the files into Ghidra\Processors creating a new subfolder C166.

IMPORTANT:
I have created these files for Ghidra 9.2.  I have no idea if they still work with newer Ghidra versions.
If it does not work download the old Ghidra version.

IMPORTANT:
I have made many changes on the SLA file.
But nevertheless there are several problems with the decompilation.
This never worked perfectly.
The resulting code has several issues.
I was not able to fix this as Ghidra file format is practically undocumented and it seems that this processor is very different from others and so there is no solution.

[C166.ZIP](https://github.com/esaulenka/Ghidra_C166/files/14526571/C166.ZIP)
